### Generate Samples

In [3]:
import numpy as np
import scipy.stats as stats


# Generate random values from Poisson distribution for East Java
np.random.seed(6)
east_java_ages1 = stats.poisson.rvs(loc=18, mu=30, size=30)
east_java_ages2 = stats.poisson.rvs(loc=18, mu=10, size=20)
east_java_ages = np.concatenate((east_java_ages1, east_java_ages2))

# Generate random values from Poisson distribution for West Java
np.random.seed(12)
west_java_ages1 = stats.poisson.rvs(loc=18, mu=33, size=30)
west_java_ages2 = stats.poisson.rvs(loc=18, mu=13, size=20)
west_java_ages = np.concatenate((west_java_ages1, west_java_ages2))

print( east_java_ages.mean() )
print( west_java_ages.mean() )

40.68
42.8


### Do Two-Sample T-Test

In [4]:
stats.ttest_ind(a= east_java_ages,
                b= west_java_ages,
                equal_var=False)    # Assume samples have equal variance false

TtestResult(statistic=-0.9745088195709343, pvalue=0.33221768005728836, df=97.41454338473416)

* pvalue = 0.33 means 33% chance we'd see sample data this far apart if the two groups tested are actually identical.
* Even if we were using a 70% confidence level we would fail to reject (accept) the null hypothesis, since the p-value is greater than the corresponding significance level of 30%. 